yt_dlp/extractor/peertube.py

```diff
@@ -1332,28 +1354,10 @@ class PeerTubePlaylistIE(InfoExtractor):
         },
         'playlist_mincount': 9,
     }]
-    _PAGE_SIZE = 30
-
-    def _call_api(self, host, uuid, path, note=None, errnote=None, fatal=True):
-        return self._download_json(
-            self._API_BASE % (host, uuid, path), uuid,
-            note=note, errnote=errnote, fatal=fatal)
-
-    def _fetch_page(self, host, uuid, page):
-        page += 1
-        video_data = self._call_api(
-            host, uuid, f'/videos?sort=-createdAt&start={self._PAGE_SIZE * (page - 1)}&count={self._PAGE_SIZE}',
-            note=f'Downloading page {page}').get('data', {})
-        for video in video_data:
-            shortUUID = try_get(video, lambda x: x['video']['shortUUID'])
-            video_title = try_get(video, lambda x: x['video']['name'])
-            yield self.url_result(
-                f'https://{host}/w/{shortUUID}', PeerTubeIE.ie_key(),
-                video_id=shortUUID, video_title=video_title)
 
     def _real_extract(self, url):
```

Comment:
Seems same as channelIE except type is hardcoded to `video-playlists`

Reply:
Should I merge it into one extractor?

Reply:
I was thinking of splitting it out into a function (say `_extract_playlist`) that takes `host, type, id`. But if merging is easier, then even better

Reply:
https://github.com/yt-dlp/yt-dlp/pull/1023/commits/58402809978be8183b7da022f7d349e8dcb99e08 is this good?